concent_api/core/tests/test_payments_sci_backend.py

```diff
@@ -0,0 +1,259 @@
+            closure_time=self.current_time,
+        )
+
+    def test_that_sci_backend_make_force_payment_to_provider_should_pay_at_least_requestor_account_balance(self):
+        with mock.patch(
+            'core.payments.payment_interface.PaymentInterface.__new__',
+            return_value=mock.Mock(
+                get_deposit_value=mock.Mock(
+                    return_value=self.transaction_value - 1
+                ),
+                force_payment=mock.Mock(
+                    return_value=MOCK_TRANSACTION_HASH
+                ),
+            )
+        ) as new_sci_rpc:
+            transaction_hash = sci_backend.make_force_payment_to_provider(
+                self.task_to_compute.requestor_ethereum_address,
+                self.task_to_compute.provider_ethereum_address,
+                self.transaction_value,
+                self.current_time,
+            )
+
+        self.assertEqual(transaction_hash, MOCK_TRANSACTION_HASH)
+
+        new_sci_rpc.return_value.get_deposit_value.assert_called_with(
+            Web3.toChecksumAddress(self.task_to_compute.requestor_ethereum_address)
+        )
+
+        new_sci_rpc.return_value.force_payment.assert_called_with(
+            requestor_address=Web3.toChecksumAddress(self.task_to_compute.requestor_ethereum_address),
+            provider_address=Web3.toChecksumAddress(self.task_to_compute.provider_ethereum_address),
+            value=self.transaction_value - 1,
+            closure_time=self.current_time,
+        )
+
+    def test_that_sci_backend_get_transaction_count_should_return_transaction_count(self):
+        with mock.patch(
+            'core.payments.payment_interface.PaymentInterface.__new__',
+            return_value=mock.Mock(
+                get_transaction_count=mock.Mock(
+                    return_value=self.transaction_count
+                ),
+            )
+        ) as new_sci_rpc:
+            transaction_count = sci_backend.get_transaction_count()
+
+        self.assertEqual(transaction_count, self.transaction_count)
+
+        new_sci_rpc.return_value.get_transaction_count.assert_called()
+
+    def test_that_sci_backend_get_deposit_value_should_return_deposit_value(self):
+        with mock.patch(
+            'core.payments.payment_interface.PaymentInterface.__new__',
+            return_value=mock.Mock(
+                get_deposit_value=mock.Mock(
+                    return_value=self.deposit_value
+                ),
+            )
+        ) as new_sci_rpc:
+            deposit_value = sci_backend.get_deposit_value(
+                self.task_to_compute.requestor_ethereum_address,
+            )
+
+        self.assertEqual(deposit_value, self.deposit_value)
+
+        new_sci_rpc.return_value.get_deposit_value.assert_called_with(
+            self.task_to_compute.requestor_ethereum_address,
+        )
+
+    def test_that_sci_backend_force_subtask_payment_should_return_transaction_hash(self):
+        PaymentInterface.__instance = None
```

Comment:
Is it needed? If so, why?

Reply:
No, it is leftover for my previous, not-working solution for mocking singleton object. It seems that if I have mocked it once during the test, then in other tests I couldn't overwrite it. I ended up mocking the whole `__new__` function instead of inner function which was called only if the instance does not already exist.